app/grandchallenge/algorithms/tasks.py

```diff
@@ -21,6 +23,100 @@
 from grandchallenge.subdomains.utils import reverse
 
 
+@shared_task
+def create_algorithm_job_for_inputs(
+    *, algorithm_image_pk, civ_pks, upload_pks, creator_pk
+):
+    try:
+        algorithm_image = AlgorithmImage.objects.get(pk=algorithm_image_pk)
+    except AlgorithmImage.DoesNotExist:
+        return
+
+    if Job.objects.filter(
+        algorithm_image=algorithm_image,
+        inputs__pk__in=civ_pks,
+        creator_id=creator_pk,
+    ).exists():
+        return
+
+    # Editors group should be able to view session jobs for debugging
+    groups = [algorithm_image.algorithm.editors_group]
+
+    job = create_algorithm_job_with_inputs(
+        algorithm_image=algorithm_image,
+        inputs=civ_pks,
+        creator=get_user_model().objects.get(pk=creator_pk),
+        extra_viewer_groups=groups,
+    )
+    start_jobs = execute_algorithm_job_for_inputs.signature(
+        kwargs={"job_pk": job.pk}, immutable=True
+    )
+    if upload_pks:
+        image_tasks = group(
+            create_component_interface_value_for_image.signature(
+                kwargs={
+                    "component_interface_value_pk": civ_pk,
+                    "upload_pk": upload_pk,
+                },
+                immutable=True,
+            )
+            for civ_pk, upload_pk in upload_pks.items()
+        )
+        start_jobs = chord(image_tasks, start_jobs)
+
+    start_jobs.apply_async()
+    return job
+
+
+@shared_task
+def create_component_interface_value_for_image(
+    *, component_interface_value_pk, upload_pk
+):
+    session = RawImageUploadSession.objects.get(pk=upload_pk)
+    session.status = RawImageUploadSession.REQUEUED
+    session.save()
+
+    build_images(upload_session_pk=upload_pk)
+
+    if session.image_set.count() > 1:
+        raise ValueError("Image imports should result in a single image")
+
+    for image in session.image_set.all():
+        civ = ComponentInterfaceValue.objects.get(
+            pk=component_interface_value_pk
+        )
+        civ.image = image
+        civ.save()
+
+
+@shared_task
+def execute_algorithm_job_for_inputs(*, job_pk):
+    job = Job.objects.get(pk=job_pk)
+
+    # Send an email to the algorithm editors and creator on job failure
+    linked_task = send_failed_jobs_email.signature(
+        kwargs={"job_pks": [job.pk]}, immutable=True
+    )
+
+    # check if all ComponentInterfaceValue's have a value.
+    # Todo: move this check to execute() code when using inputs is done
```

Comment:
TODO

Reply:
Added it here so the old way keeps working, even when another input has been added to the algorithm. So other inputs are only required when using the new form. Do you want me to change that?